test/widgets/theme_test.dart

```diff
@@ -0,0 +1,78 @@
+import 'package:checks/checks.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:zulip/widgets/text.dart';
+import 'package:zulip/widgets/theme.dart';
+
+import '../flutter_checks.dart';
+
+void main() {
+  group('button text size and letter spacing', () {
+    Future<void> doCheck(
+      String description, {
+      required Widget Function(BuildContext context, String text) buttonBuilder,
+      double? ambientTextScaleFactor,
+    }) async {
+      testWidgets(description, (WidgetTester tester) async {
+        if (ambientTextScaleFactor != null) {
+          tester.platformDispatcher.textScaleFactorTestValue = ambientTextScaleFactor;
+          addTearDown(tester.platformDispatcher.clearTextScaleFactorTestValue);
+        }
+        const buttonText = 'Zulip';
+        double? expectedFontSize;
+        double? expectedLetterSpacing;
```

Comment:
Can tighten up expectations here slightly:
```suggestion
        late final double expectedFontSize;
        late final double expectedLetterSpacing;
```

That way it's clear (and enforced) that these only get set once, and that they always get set before they're read.